### PR TITLE
MAYA-124418 copy transform to the correct variant.

### PR DESCRIPTION
**Pull request by pierrebai-adsk**
- Need to find and temporarily switch to the variant that contains the Maya Reference.
- Since variant set and variant name are under the user control, we need an explicit search.
- Copy only the transform and properly handle the case of the source and destination not being variants.

Second version added:
- Move the copying of the Maya Reference transform to the edit router.
- Go back to using mergePrim to do the work.
- Properly handle the case teh destination is in a variant.
- Cleanup the edit router code by refactor common code and common code patterns.
- Read all expected context arguments in one go, makes it easeier to understand what argument are expected.
- Move inner lambda function to a separate function for code clarity.
- Avoiding having such a function within a function makes the code more linear and easier to read.

### CODE REVIEW

**Comment by pierrebai-adsk** [comment]
pr-build!

File: lib/usd/translators/mayaReferenceUpdater.cpp [hunk 1]
```diff
@@ -177,6 +176,19 @@ bool PxrUsdTranslators_MayaReferenceUpdater::editAsMaya()
     return true;
 }
 
+bool callEditRouter(
```
Comment (pierrebai-adsk) [inline lib/usd/translators/mayaReferenceUpdater.cpp:179]: Will be useful if we change the edit router API to return a success/failure value or if we allow explicitly to throw exception on error, we will be able to catch the exception here and convert it to a success/failure value.

File: lib/usd/translators/mayaReferenceUpdater.cpp [hunk 2]
```diff
@@ -191,67 +203,53 @@ UsdMayaPrimUpdater::PushCopySpecs PxrUsdTranslators_MayaReferenceUpdater::pushCo
...
 
-    // Use the edit router to find the destination layer and path.
-    auto dstEditRouter = MayaUsd::getEditRouter(TfToken("mayaReferencePush"));
-    if (!dstEditRouter) {
-        return PushCopySpecs::Failed;
-    }
-
     PXR_NS::VtDictionary routerContext = getContext()->GetUserArgs();
+
+    // Pass the source and destination stage, layer and path to routers.
```
Comment (pierrebai-adsk) [inline lib/usd/translators/mayaReferenceUpdater.cpp:208]: These values are required for the copy-transform code. In general, an edit router might want to know these detail to make decision where to route.

**Review by ppt-adsk — APPROVED** [review]
Nice cleanup, thanks for the fix.

**Pull request merged**